aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -157,20 +146,51 @@ module AptosFramework::Token {
         uri: ASCII::String,
     }
 
-    public fun token_id<TokenType: copy + drop + store>(token: &Token<TokenType>): &ID {
-        &token.id
+    // Some Tokens may want additional fields outside the specification. As the Token itself does
+    // not contain the data, it is easier in Move to have this extra metadata sit on the side so
+    // that only creation and read operations (outside of Move) need to deal with the nuance of 
+    // metadata.
+    struct TokenMetadata<TokenType: store> has key {
+        metadata: Table<ID, TokenType>,
     }
 
-    public fun create_token_script<TokenType: copy + drop + store>(
+    fun initialize_token_metadata<TokenType: store>(account: &signer) {
```

Comment:
how do we look up the data given a token? there's no address for the metadata specified in the token. (same for collections). probably need to attach creator address

Reply:
The ID contains the creators address